hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/HdfsScheduler.java

```diff
@@ -339,6 +339,15 @@ private String getNextTaskName(String taskType) {
   private ExecutorInfo createExecutor(String taskIdName, String nodeName, String executorName,
     List<Resource> resources) {
     int confServerPort = hdfsFrameworkConfig.getConfigServerPort();
+
```

Comment:
This is a little bit strange string concatentation. It should be like this so the concatenation is done consistently "export JAVA_HOME=$MESOS_DIRECTORY/" + hdfsFrameworkConfig.getJreVersion() + "&& env ; cd hdfs-mesos-* && "
+      + "exec `if [ -z \"$JAVA_HOME\" ]; then echo java; "
+      + "else echo $JAVA_HOME/bin/java; fi` "
+      + "$HADOOP_OPTS "
+      + "$EXECUTOR_OPTS "
+      + "-cp lib/*.jar org.apache.mesos.hdfs.executor." + executorName;